contracts/btc-finality/src/finality.rs

```diff
@@ -632,14 +632,20 @@ pub fn distribute_rewards(deps: &mut DepsMut, env: &Env) -> Result<(), ContractE
     };
     // Get the voting power of the active FPS
     let total_voting_power = active_fps.iter().map(|fp| fp.power as u128).sum::<u128>();
-    // Get the rewards to distribute (bank balance of the finality contract)
+    // Get the rewards to distribute (bank balance of the finality contract minus already distributed rewards)
+    let distributed_rewards = TOTAL_REWARDS.load(deps.storage)?;
```

Comment:
Is it possible to get the total reward by querying the contract's balance, rather than tracking it in a Kv store?

Reply:
The problem with this is, that `TOTAL_REWARDS` are the rounded sum of the distributed rewards over finality providers. They are not necessarily the same. With this impl, we left the remainder in the balance, and it'll be distributed later eventually. Better than send an amount that doesn't match the sum of the distributed rewards, IMO.